File: javascript/triangle/triangle.js
function Triangle(a, b, c) {
  this.a = a;
  this.b = b;
  this.c = c;

  this.kind = function () {
    if ((this.a == 0 && this.b == 0 && this.c == 0) || this.a < 0 || this.b < 0 || this.c < 0) {
      throw new Error();
    }
    if ((this.a + this.b < this.c) || (this.b + this.c < this.a) || (this.c + this.a < this.b)) {
      throw new Error();
    }
    kindName = "";
    if (this.a == this.b && this.b == this.c) {
      kindName = "equilateral";
    }
    else if (this.a == this.b || this.b == this.c || this.a == this.c) {
      kindName = "isosceles";
    } else {
      kindName = "scalene";
    }
    return kindName;
  };
}

module.exports = Triangle;

File: csharp/triangle/Triangle.cs
enum TriangleKind
{
  Equilateral,
  Isosceles,
  Scalene
}

class TriangleException : System.Exception
{

}

class Triangle
{
  public static TriangleKind Kind(decimal a, decimal b, decimal c)
  {
    if (a == 0 && b == 0 && c == 0)
      throw new TriangleException();
    if (a < 0 || b < 0 || c < 0)
      throw new TriangleException();
    if ((a + b <= c) || (a + c <= b) || (b + c <= a))
      throw new TriangleException();
    if (a == b && b == c)
      return TriangleKind.Equilateral;
    else if (a == b || b == c || a == c)
      return TriangleKind.Isosceles;
    return TriangleKind.Scalene;
  }
}